### DISCUSSION TITLE
2 Builds on Alpine on one page, cannot access $store ?

### DISCUSSION THREAD

**Discussion by yabdab** [alpinejs/alpine#4331] [1. Help]
So I have a page with a traditional build of Alpine and one built with a prefix.

I have a Alpine.store that works just fine when the prefixed version in on page alone. But when I add the traditional build to the page, I can no longer access the store data.

For example I am using code like this that can no longer see. the "items" ...
`Alpine.store("formsnap").items.forEach((item) => { ... });`

Is there a special way to target the "Alpine" instance  using the prefix?

Or should I import the prefixed version. in with different name like so ...
`import prefixedAlpine from "alpinejs";`

Thanks to any help folks can give.

**Comment by ekwoka** [comment]
The cdn link would override the Alpine window instance (or vice versa).

Why do you need two on the page?

But to differentiate, you'd need to just always use your imported one (or passed to the plugin functions the one you want applied)

**Reply by yabdab** [reply]
I am writing an add-on product that build web sites. Just trying to shore up conflicts in case the user add their own copy/version of Alpine in the same page.

How can I  make sure I am using the version I imported? Is there a way to target it some way specifically?

**Reply by yabdab** [reply]
So looks like I fixed it by changing this ...

`Alpine.store("formsnap").items.forEach((item) => { ... });`

to this ...

`this.$store.formsnap.items.forEach((item) => { ... });`

**Reply by ekwoka** [reply]
The easy way is do not add yours to the window, and do not ever use the one from the window.

Always pass your isntance to anything you need to use.

**Reply by yabdab** [reply]
@ekwoka What would be best way to "pass your isntance" ? Really appreciate your help.

**Reply by ekwoka** [reply]
Like when you import Alpine, and do your stuff.

Pass it to the things that need it.

You can setup your code as Alpine Plugin callbacks that you put into Alpine.plugin.

If they only use that Alpine then it all works fine :)

**Reply by yabdab** [reply]
OK, I get it, and that works. BUT I have other children components ( within the main ) that need to use the parent's functions as well. So every other script using `Alpine.data("childComponent")` fails because it cannot find `Alpine`. I cannot add the children code to main build because it is a modular setup and the children are added as needed.

Would another option be to rename "Alpine" to "MyAlpine" and attach that to the window?

**Reply by ekwoka** [reply]
What do you mean use the parents functions?

Child components has parents methods in their scope on `this`.

And you pass all of them your Alpine, so idk what you mean by "cannot find Alpine"? You give it to them. Then they have it.

**Reply by yabdab** [reply]
I have a main x-data parent object.

That is included in my build of Alpine, so that part works.

Now if I add(nest) a child x-data within the parent  , which is not inline ( it is a separate script that is called ), then the child cannot see the Alpine instance from the parent.


`
-parent 
|
|--child 1
|--child 2
|
--
`

The child script looks like so ...

```
 Alpine.data("childDataItem", () => ({ ... }));
```

Hope that explains it.

**Reply by ekwoka** [reply]
> Now if I add(nest) a child x-data within the parent , which is not inline ( it is a separate script that is called ), then the child cannot see the Alpine instance from the parent.

Yes it can.

on `this` within functions.

**Reply by yabdab** [reply]
"Alpine" is not defined in child components.🤷‍♂️

**Reply by ekwoka** [reply]
So use `this`

(but alpine should still be defined, since you made the component somehow and you can pass it Alpine before defining it)

Additionally, you cannot do `Alpine.data('componentName')` to get access to instances of that component.
